stl/inc/xmemory

```diff
@@ -776,6 +776,21 @@ _NODISCARD constexpr allocation_result<typename allocator_traits<_Alloc>::pointe
 // The number of user bytes a single byte of ASAN shadow memory can track.
 _INLINE_VAR constexpr size_t _Asan_granularity = 8;
 
+_NODISCARD inline const void* _Get_asan_aligned_first(const void* _First, const void* _Last) noexcept {
+    const auto _First_address     = reinterpret_cast<uintptr_t>(_First);
+    const auto _Last_address       = reinterpret_cast<uintptr_t>(_Last);
+
+    const auto _Aligned_address = (_First_address + _Asan_granularity - 1) & ~(_Asan_granularity - 1);
+
+    if (_Aligned_address > _Last_address) {
+        return nullptr;
+    } else {
+        // strict provenance requires that we reconstruct `_Aligned_address` via `_First`

```

Comment:
"Strict [pointer] provenance"? Why do we care? We've never supported an implementation with strict pointer provenance, and the pertinent parts of the Standard have been removed. Is this actually an issue for ASAN?

Reply:
It is not, but I was nervous about pointer provenance and I don't know the exact rules -.-

Reply:
I don't either - they don't come up often. C++17 [basic.stc.dynamic.safety]/3.4 says we can perform arithmetic in the `uintptr_t` domain if the result converted back to a pointer would compare equal to a safely-derived pointer computable in the pointer domain. So we could simply return the result of reinterpreting `_Aligned_address` even on an implementation with strict pointer safety (which again, ours is not).